include/seqan3/alphabet/union_alphabet.hpp

```diff
@@ -81,7 +81,7 @@ constexpr auto alphabet_prefix_sum_sizes()
 
     array_t prefix_sum{0, alphabet_types::value_size...};
     for (auto i = 1u; i < N; ++i)
-        prefix_sum[i] += prefix_sum[i-1];
+        prefix_sum[i] = static_cast<rank_t>(prefix_sum[i] + prefix_sum[i-1]);
```

Comment:
Yes. Adding two `char`s results in an rvalue of type `int`, which we then store in `prefix_sum[i]` which is of type `char`.